core/services/versionchooser/utils/chooser.py

```diff
@@ -130,6 +130,49 @@ async def set_version(self, image: str, tag: str) -> web.StreamResponse:
                 logging.critical("Error: %s: %s", type(error), error)
                 return web.Response(status=500, text=f"Error: {type(error)}: {error}")
 
+    async def delete_version(self, image: str, tag: str) -> web.StreamResponse:
+        """Deletes the selected version.
+
+        Args:
+            image (str): the repository of the image
+            tag (str): the desired tag
+
+        Returns:
+            web.Response:
+                200 - OK
+                400 - Invalid image/tag
+                403 - image cannot be deleted
+                500 - Internal error (unable to read config file/docker refused to delete image)
+        """
+        full_name = f"{image}:{tag}"
+        # refuse if it is the current image
+        with open(DOCKER_CONFIG_PATH, "r+") as startup_file:
+            try:
+                core = json.load(startup_file)["core"]
+                current_tag = core["tag"]
+                current_image = core["image"]
+                if image == current_image and tag == current_tag:
+                    return web.Response(status=403, text=f"Refusing to delete the image in use ({image}:{tag})")
+            except Exception as e:
+                logging.warning(f"Unable to read config file: {e}")
+                return web.Response(status=500, text=f"Unable read config file: {e}")
+
+        # check if image exists
+        image = self.client.images.get(full_name)
+        if not image:
+            logging.warning(f"Image not found: {full_name}")
+            return web.Response(status=404, text=f"image not found: {full_name}")
+
+        # actually attempt to delete it
+        logging.info(f"Deleting image {image}:{tag}...")
+        try:
+            self.client.images.remove(full_name, force=False, noprune=False)
+            logging.info("Image deleted successfully")
```

Comment:
I think "Image successfully deleted." is the correct form.

Reply:
"succesfully" is an adverb of mode. Same as in portuguese, [they can be placed before and after the verb](https://www.twinkl.com.br/teaching-wiki/adverb-of-manner#:~:text=An%20adverb%20of%20manner%20modifies,the%20reader%20a%20clearer%20picture.).
This way sounds better to me.